azkaban-web-server/src/main/java/azkaban/flowtrigger/quartz/FlowTriggerScheduler.java

```diff
@@ -0,0 +1,207 @@
+            this.scheduler
+                .registerJob(flowTrigger.getSchedule().getCronExpression(), new QuartzJobDescription
+                    (FlowTriggerQuartzJob.class, generateGroupName(flow), contextMap));
+          }
+        } catch (final IOException ex) {
+          logger.error("error in getting flow file", ex);
+        } finally {
+          FlowLoaderUtils.cleanUpDir(tempDir);
+        }
+      }
+    }
+  }
+
+  /**
+   * Retrieve the list of scheduled flow triggers from quartz database
+   */
+  public List<ScheduledFlowTrigger> getScheduledFlowTriggerJobs() {
+    final Scheduler quartzScheduler = this.scheduler.getScheduler();
+    try {
+      final List<String> groupNames = quartzScheduler.getJobGroupNames();
+
+      final List<ScheduledFlowTrigger> flowTriggerJobDetails = new ArrayList<>();
+      for (final String groupName : groupNames) {
+        final JobKey jobKey = new JobKey(QuartzScheduler.DEFAULT_JOB_NAME, groupName);
+        ScheduledFlowTrigger scheduledFlowTrigger = null;
+        try {
+          final JobDetail job = quartzScheduler.getJobDetail(jobKey);
+          final JobDataMap jobDataMap = job.getJobDataMap();
+
+          final String flowId = jobDataMap.getString(FlowTriggerQuartzJob.FLOW_ID);
+
+          final String projectJson = jobDataMap.getString(FlowTriggerQuartzJob.PROJECT);
+          final Project project = FlowUtils.toProject(projectJson);
+          final FlowTrigger flowTrigger = (FlowTrigger) jobDataMap
+              .get(FlowTriggerQuartzJob.FLOW_TRIGGER);
+          final String submitUser = jobDataMap.getString(FlowTriggerQuartzJob.SUBMIT_USER);
+          final List<? extends Trigger> quartzTriggers = quartzScheduler.getTriggersOfJob(jobKey);
+          scheduledFlowTrigger = new ScheduledFlowTrigger(
+              project.getName(),
+              flowId, flowTrigger, submitUser, quartzTriggers.isEmpty() ? null
+              : quartzTriggers.get(0));
+        } catch (final Exception ex) {
+          logger.error(String.format("unable to get flow trigger by job key %s", jobKey), ex);
+          scheduledFlowTrigger = null;
+        }
+
+        flowTriggerJobDetails.add(scheduledFlowTrigger);
+      }
+      return flowTriggerJobDetails;
+    } catch (final SchedulerException ex) {
+      logger.error("unable to get scheduled flow triggers", ex);
+      return new ArrayList<>();
+    }
+  }
+
+  /**
+   * Unschedule all possible flows in a project
+   */
+  public void unscheduleAll(final Project project) throws SchedulerException {
+    for (final Flow flow : project.getFlows()) {
+      logger.info("unscheduling flow" + flow.getProjectId() + "." + flow.getId() + " if it has "
+          + " schedule");
+      this.scheduler.unregisterJob(generateGroupName(flow));
+    }
+  }
+
+  private String generateGroupName(final Flow flow) {
+    return String.valueOf(flow.getProjectId()) + "." + flow.getId();
+  }
+
+  public void start() {
```

Comment:
We should not need to start or shutdown here.

After sync-up, I still believe we should keep the original Quartz Scheduler a separate module. You don't need to handle start or shutdown API, which should be handled by AZ webserver, rather than plugin. Besides, you don't and shouldn't call all APIs like Pause or checkExists, which should be `QuartzScheduler`'s responsibility. This class should just call its API. It will be hard to evolve or enhance `QuartzScheduler` if we go this direction.